File: 16_online_polling_system/python/online_polling_system.py
poll = {
    "Option A": 0,
    "Option B": 0,
    "Option C": 0
}

while True:
    print("\n1. Vote")
    print("2. View Results")
    print("3. Exit")

    choice = input("Choice: ")

    if choice == "1":
        print("\nOptions:")
        for i, option in enumerate(poll.keys(), start=1):
            print(i, option)

        vote = int(input("Select option number: "))
        options = list(poll.keys())

        if 1 <= vote <= len(options):
            poll[options[vote - 1]] += 1
            print("Vote recorded.")
        else:
            print("Invalid option.")

    elif choice == "2":
        print("\nResults:")
        for option, count in poll.items():
            print(option, ":", count)

    elif choice == "3":
        break

    else:
        print("Invalid choice.")

File: 16_online_polling_system/java/Online_pollingSystem.java
import java.util.*;

public class Online_pollingSystem {

    public static void main(String[] args) {

        Map<String, Integer> poll = new LinkedHashMap<>();
        poll.put("Option A", 0);
        poll.put("Option B", 0);
        poll.put("Option C", 0);

        Scanner sc = new Scanner(System.in);

        while (true) {
            System.out.println("\n1. Vote");
            System.out.println("2. View Results");
            System.out.println("3. Exit");
            System.out.print("Choice: ");

            String choice = sc.nextLine();

            if (choice.equals("1")) {
                int i = 1;
                for (String option : poll.keySet()) {
                    System.out.println(i + ". " + option);
                    i++;
                }

                System.out.print("Select option number: ");
                int vote = Integer.parseInt(sc.nextLine());
                List<String> options = new ArrayList<>(poll.keySet());

                if (vote >= 1 && vote <= options.size()) {
                    String selected = options.get(vote - 1);
                    poll.put(selected, poll.get(selected) + 1);
                    System.out.println("Vote recorded.");
                } else {
                    System.out.println("Invalid option.");
                }

            } else if (choice.equals("2")) {
                System.out.println("\nResults:");
                for (Map.Entry<String, Integer> entry : poll.entrySet()) {
                    System.out.println(entry.getKey() + " : " + entry.getValue());
                }

            } else if (choice.equals("3")) {
                sc.close();
                return;

            } else {
                System.out.println("Invalid choice.");
            }
        }
    }
}

